### PR TITLE
Understanding the API docs

### PR DESCRIPTION
**Pull request by sfarrens**
## Summary

- added a age to help users/developers understand the API docs
- cleaned up some minor points in the docs

## Reviewer Checklist

> Reviewers should tick the following boxes before approving and merging the PR.

- [x] The PR targets the `develop` branch
- [x] The PR is assigned to the developer
- [x] The PR has appropriate labels
- [x] The PR is included in appropriate projects and/or milestones
- [x] The PR includes a clear description of the proposed changes
- [x] If the PR addresses an open issue the description includes "closes #<ISSUE NUMBER>"
- [x] The code and documentation style match the current standards
- [x] Documentation has been added/updated consistently with the code
- [x] All CI tests are passing
- [x] API docs have been built and checked at least once (if relevant)
- [x] All changed files have been checked and comments provided to the developer
- [x] All of the reviewer's comments have been satisfactorily addressed by the developer

### CODE REVIEW

File: docs/source/configuration.md [hunk 1]
```diff
@@ -103,7 +103,8 @@ ShapePipe will look for files of the form `galaxy-00-0.fits`,
 `galaxy-00-1.fits`, etc. in the directory `/home/username/my_input_dir` and
 save outputs to `/home/username/my_output_dir`. Note that the `FILE_PATTERN`
 does not need to be complete, in other words files of the form
-`mygalaxy-00-0.fits` would equally be found with the above options.
+`mygalaxy-00-0.fits` would equally be found if `CORRECT_FILE_PATTERN = True`
+was added to the options above.
```
Comment (lbaumo) [inline docs/source/configuration.md:107]: was should be were

File: docs/source/understanding_api.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+and function in the package does. For example, what inputs they expect, what
+outputs they provide, and what the various options do. These pages are
+automatically generated from docstrings (i.e. the sections starting and ending with three double quotes `"""`) written in the code.
+
+## Standard API Docs
+
+All the classes/functions include a short description of what they do. This is followed by a `Parameters` section containing a bullet point list of the expected input arguments. For each parameter you will see in brackets the expected input type (e.g. `int`, `float`, `list`, etc.) followed by a brief description of what the argument is for. Parameters listed as *optional* in the brackets do not need to be provided and will default to some predefined value.
+
+```{note}
+If an optional argument does not explicitly specify the default parameter value then the user should expect that this means the default will be `None`, `''`, `[]`, etc. depending on the input data type.
```
Comment (lbaumo) [inline docs/source/understanding_api.md:22]: this renders very weirdly, and the reader has to scroll horizontally to read the entire sentence.

File: docs/source/understanding_api.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+Some objects may contain a `Notes` section providing further details for the class/function. Some objects may also contain a `See Also` section that links to another related object.
+
+Finally, all objects include `[source]` button that will allow you to view the  code implementation for the given class/function. You can switch back to the API docs by clicking the corresponding `[docs]` button.
+
+## ShapePipe Module Docs
+
+In addition to the stand API docs, we provide a description of each ShapePipe module defined in the `__init__.py` file for each module package. On these pages you will find a header that specifies the package author, the `Parent module` (i.e. a module that should be run before the module in question), and a brief description of the inputs and outputs. This is followed by a description of the module. Finally, there is a section that defines all of the module-specific [config file options](configuration.md). Each option is listed with the exact name that should be added to the config file, followed by the expected value type. Underneath this is a brief description of what this option does. For example, the following option for a module called `my_module`
```
Comment (lbaumo) [inline docs/source/understanding_api.md:35]: end with :

File: docs/source/understanding_api.md (hunk 2 above)
Comment (sfarrens) [reply docs/source/understanding_api.md]: OK, should be OK now

File: docs/source/understanding_api.md (hunk 3 above)
Comment (sfarrens) [reply docs/source/understanding_api.md]: Done

**Pull request merged**